website/server/controllers/api-v3/shops.js

```diff
@@ -0,0 +1,121 @@
+import { authWithHeaders } from '../../middlewares/api-v3/auth';
+import shops from '../../../../common/script/libs/shops';
+
+let api = {};
+
+/**
+ * @apiIgnore
+ * @api {get} /api/v3/shops/market get the available items for the market
+ * @apiVersion 3.0.0
+ * @apiName GetMarketItems
+ * @apiGroup Shops
+ *
+ * @apiSuccess {Object} data List of push devices
+ * @apiSuccess {string} message Success message
+ */
+api.getMarketItems = {
+  method: 'GET',
+  url: '/shops/market',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    let resObject = {
+      identifier: 'market',
+      text: res.t('market'),
+      notes: res.t('welcomeMarketMobile'),
+      imageName: 'npc_alex',
+      categories: shops.getMarketCategories(user, req.language),
+    };
+
+    res.respond(200, resObject);
+  },
+};
+
+/**
+ * @apiIgnore
+ * @api {get} /api/v3/shops/quests get the available items for the quests shop
+ * @apiVersion 3.0.0
+ * @apiName GetQuestShopItems
+ * @apiGroup Shops
+ *
+ * @apiSuccess {Object} data List of push devices
+ * @apiSuccess {string} message Success message
+ */
+api.getQuestShopItems = {
+  method: 'GET',
+  url: '/shops/quests',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    let resObject = {
+      identifier: 'questShop',
+      text: res.t('quests'),
+      imageName: 'npc_ian',
+      categories: shops.getQuestShopCategories(user, req.language),
+    };
+
+    res.respond(200, resObject);
+  },
+};
+
+/**
+ * @apiIgnore
+ * @api {get} /api/v3/shops/time-travelers get the available items for the time travelers shop
+ * @apiVersion 3.0.0
+ * @apiName GetTimeTravelersShopItems
+ * @apiGroup Shops
+ *
+ * @apiSuccess {Object} data List of push devices
+ * @apiSuccess {string} message Success message
+ */
+api.getTimeTravelerShopItems = {
+  method: 'GET',
+  url: '/shops/time-travelers',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    let resObject = {
+      identifier: 'timeTravelersShop',
+      text: res.t('timeTravelers'),
+      notes: user.purchased.plan.consecutive.trinkets > 0 ? res.t('timeTravelersPopover') : res.t('timeTravelersPopoverNoSubMobile'),
+      imageName: user.purchased.plan.consecutive.trinkets > 0 ? 'npc_timetravelers_active' : 'npc_timetravelers',
```

Comment:
user.purchased.plan.consecutive.trinkets > 0 should go in a variable since it's used multiple times